app/home/directives/home-story.component.ts

```diff
@@ -0,0 +1,108 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {DatePipe} from '@angular/common';
+import {ROUTER_DIRECTIVES} from '@angular/router';
+
+import {ImageLoaderComponent} from '../../shared/image/image-loader.component';
+import {DurationPipe} from '../../shared/file/duration.pipe';
+import {StoryModel} from '../../storyedit/models/story.model';
+
+@Component({
+  directives: [ImageLoaderComponent, ROUTER_DIRECTIVES],
+  pipes: [DatePipe, DurationPipe],
+  selector: 'home-story',
+  styleUrls: ['app/home/directives/home-story.component.css'],
+  template: `
+    <template [ngIf]="isPlusSign">
+      <a [routerLink]="editLink" class="plus-sign">
+        <i class="icon-plus"></i>
+        <p>New Story</p>
+      </a>
+    </template>
+    <template [ngIf]="!isPlusSign">
+      <a [routerLink]="editLink">
+        <image-loader *ngIf="storyImage" [src]="storyImage"></image-loader>
+        <div *ngIf="!storyImage" class="no-image">No Image</div>
+      </a>
+      <h2>
+        <a *ngIf="storyTitle" [routerLink]="editLink">{{storyTitle}}</a>
+        <a *ngIf="!storyTitle" [routerLink]="editLink">(Untitled)</a>
+      </h2>
+      <p class="duration">{{storyDuration | duration}}</p>
+      <p class="modified">{{storyUpdated | date:"MM/dd/yy"}}</p>
+      <p *ngIf="statusClass" [class]="statusClass">{{statusText}}</p>
+    </template>
+  `
+})
+
+export class HomeStoryComponent implements OnInit {
+
+  @Input() story: StoryModel;
+
+  editLink: any[];
+  isPlusSign: boolean;
+
+  storyId: number;
+  storyImage: string;
+  storyTitle: string;
+  storyDuration: number;
+  storyUpdated: Date;
+
+  statusClass: string;
+  statusText: string;
+
+  ngOnInit() {
+    this.storyId = this.story.id;
+    this.isPlusSign = this.story.isNew && !this.story.isStored();
+    this.storyTitle = this.story.title;
+    this.storyUpdated = this.story.lastStored || this.story.updatedAt;
+
+    if (this.story.isNew) {
+      this.editLink = ['/create'];
+      if (this.story.parent) {
+        this.editLink.push(this.story.parent.id);
+      }
+    } else {
+      this.editLink = ['/edit', this.story.id];
+    }
+
+    // TODO: draft audios
```

Comment:
Related to the lambda stuff I'm working on.  Some way to access/playback the "temporary" upload mp3s, before the lambda-fixer-thing has a chance to move the files to the permanent location.